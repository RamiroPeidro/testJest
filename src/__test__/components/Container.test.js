import React from 'react';
import { render, act } from '@testing-library/react';
import { PhotoContext } from '../../context/PhotoContext';
import Container from '../../components/Container';

describe('Container', () => {

    it('Llama a runSearch con la propiedad searchTerm', () => {
        const mockRunSearch = jest.fn();
        const searchTerm = "test";
        
        render(
            <PhotoContext.Provider value={{ images: [], loading: true, runSearch: mockRunSearch }}>
                <Container searchTerm={searchTerm} />
            </PhotoContext.Provider>
        );

        expect(mockRunSearch).toHaveBeenCalledWith(searchTerm);
    });

    it('Cuando loading es true, se muestra el componente de carga', () => {
        const { queryByTestId } = render(
            <PhotoContext.Provider value={{ images: [], loading: true, runSearch: jest.fn() }}>
                <Container searchTerm="test" />
            </PhotoContext.Provider>
        );
        expect(queryByTestId('loader')).toBeTruthy();
    });

    it('Cuando loading es false, se muestra la galerÃ­a', () => {
        const mockImages = ['img1', 'img2', 'img3'];
        const { queryByTestId } = render(
            <PhotoContext.Provider value={{ images: mockImages, loading: false, runSearch: jest.fn() }}>
                <Container searchTerm="test" />
            </PhotoContext.Provider>
        );
        expect(queryByTestId('gallery')).toBeTruthy();
    });

    it('Coincide el Snapshot', () => {
        const { asFragment } = render(
            <PhotoContext.Provider value={{ images: [], loading: true, runSearch: jest.fn() }}>
                <Container searchTerm="test" />
            </PhotoContext.Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
