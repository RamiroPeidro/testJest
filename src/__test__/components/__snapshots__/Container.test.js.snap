// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Coincide el Snapshot 1`] = `
<DocumentFragment>
  <div
    class="photo-container"
  >
    <div
      class="loader"
      data-testid="loader"
    />
  </div>
</DocumentFragment>
`;
